test/jdk/java/awt/regtesthelpers/PassFailJFrame.java

```diff
@@ -158,9 +224,61 @@ public static void getFailureReason() {
 
         jPanel.add(okayBtnPanel, BorderLayout.SOUTH);
         dialog.add(jPanel);
-        dialog.setLocationRelativeTo(null);
+        dialog.addWindowListener(new WindowAdapter() {
+            @Override
+            public void windowClosing(WindowEvent e) {
+                super.windowClosing(e);
+                testFailedReason = "User closed the " +
+                        "dialog";
```

Comment:
I see no reason to break this string, it fits into 80-column.

Yet it's not an error: I'd rather leave the failure reason empty. In this case, the window listener can be dropped.

You can safely remove `super.windowClosing(e);`, the implementation in `WindowAdapter` does nothing.